core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -486,18 +487,24 @@ private StartupConsistencyChecker(Long fileId) {
             inconsistentUris.add(parentUri);
           }
           for (Inode childInode : ((InodeDirectory) parentInode).getChildren()) {
-            AlluxioURI childUri = parentUri.join(childInode.getName());
-            if (childInode.isDirectory()) {
-              dirsToCheck.add(childInode.getId());
-            } else {
-              childInode.lockRead();
-              try {
+            try {
+              childInode.lockReadAndCheckParent(parentInode);
+            } catch (InvalidPathException e) {
+              // This should be safe, continue.
+              LOG.debug("A file scheduled for consistency check was moved before the check.");
```

Comment:
The startup check does not block until complete, so the user can possibly do other write operations while the startup check is running. We cannot log the path because we do not have a lock on the inode (childInode.getName is not guaranteed to be accurate). We could log the parent directory, but I don't think this is would be helpful.